docs/protocol.html

```diff
@@ -143,7 +138,13 @@ <h5 class="anchor-heading"><a id="api_versions" class="anchor-link"></a><a href=
         Kafka protocol requests may take place on an SSL listener before the SSL handshake is finished). If this is considered to
         leak information about the broker version a workaround is to use SSL with client authentication which is performed at an
         earlier stage of the connection where the <code>ApiVersionRequest</code> is not available. Also, note that broker versions older
-        than 0.10.0.0 do not support this API and will either ignore the request or close connection in response to the request.</li>
+        than 0.10.0.0 do not support this API and will either ignore the request or close connection in response to the request. Also
+        note that if the client <code>ApiVersionsRequest</code> version is unsupported by the broker (client is ahead), and the broker
+        version is 2.5.0 or greater, then the broker will respond with a version 0 <a href="#The_Messages_ApiVersions">ApiVersionsResponse</a>
```

Comment:
I see it's 2.4 [here](https://github.com/apache/kafka/blob/f87c04854bc532247542f26af19198eb879c2ab6/clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsRequest.java#L120). Which one is correct?

Reply:
I believe the comment is correct. I found 2.5.0 via the [2.5.0 release notes](https://archive.apache.org/dist/kafka/2.5.0/RELEASE_NOTES.html). This is because the JIRA ticket was partially implemented in 2.4.0 and finished in 2.5.0:

```
robeyoun@robeyoun-thinkpadp16vgen1:~/development/upstream/kafka$ git log 2.5.0 --grep KIP-511 --oneline
ae38315895 KAFKA-8855; Collect and Expose Client's Name and Version in the Brokers (KIP-511 Part 2) (#7749)
f98013cc50 Part 1 of KIP-511: Collect and Expose Client's Name and Version in the Brokers #7381
robeyoun@robeyoun-thinkpadp16vgen1:~/development/upstream/kafka$ git log 2.4.0 --grep KIP-511 --oneline
f98013cc50 Part 1 of KIP-511: Collect and Expose Client's Name and Version in the Brokers #7381
```

the relevant change to the ApiVersions response arrived in 2.4

Reply:
pushed up